test/integration/smoke/test_primary_storage.py

```diff
@@ -363,6 +363,151 @@ def test_01_add_primary_storage_disabled_host(self):
         return
 
 
+class TestPrimaryStorageScope(cloudstackTestCase):
```

Comment:
Not sure if this should be in a new file. Others can comment.
The problem with adding new classes to the same test file is that marvin may run all test cases in the file on a single failure

Reply:
I see your point and want to add that for maintainability it is preferable to have a single test class per file.

Reply:
Moved the testcase to a new file.